pkg/apis/serving/v1alpha1/kfservice_status.go

```diff
@@ -73,6 +73,12 @@ func (ss *KFServiceStatus) PropagateDefaultConfigurationStatus(defaultConfigurat
 // PropagateCanaryConfigurationStatus propagates the canary Configuration status and applies its values
 // to the Service status.
 func (ss *KFServiceStatus) PropagateCanaryConfigurationStatus(canaryConfigurationStatus *knservingv1alpha1.ConfigurationStatus) {
+	// reset status if canaryConfigurationStatus is nil
+	if canaryConfigurationStatus == nil {
+		ss.Canary = StatusConfigurationSpec{}
+		conditionSet.Manage(ss).MarkUnknown(CanaryPredictorReady, "CanarySpecUnavailable", "Canary spec unavailable")
```

Comment:
Rather than exposing this, I'd much prefer to simply remove from the set. If this is not possible right now, we should cut a bug to update Knative/PKG with this. 

Keep in mind, we may potentially have tons of these conditions (i.e. CanaryExplainer, Explainer, CanaryPreprocessor, etc....)

Reply:
As stated in https://github.com/kubeflow/kfserving/issues/276, I have already added fix(implemented new function ClearCondition) to knative/pkg master branch. However, we can't pull the fix now as master branch of knative/pkg is not compatible with release-0.7 branch which we are using. We have to wait till release 0.8 to get the fix and we can change it to ClearConditions then.